src/models/nodeChain.ts

```diff
@@ -0,0 +1,120 @@
+import {
```

Comment:
This code is mostly unchanged, just moved into its own file (because of the 1 class per file rule)